renderdoc/driver/gl/wrappers/gl_draw_funcs.cpp

```diff
@@ -4445,7 +4445,29 @@ bool WrappedOpenGL::Serialise_glClear(SerialiserType &ser, GLbitfield mask)
       if(mask & (eGL_DEPTH_BUFFER_BIT | eGL_STENCIL_BUFFER_BIT))
         draw.flags |= DrawFlags::ClearDepthStencil;
 
-      AddDrawcall(draw, true);
+      ResourceId res_id;
+      auto recordResourceUses = [&res_id, this](const ContextPair &ctx, GLenum type, GLuint attachment) {
```

Comment:
I prefer not using lambdas unless it's strictly necessary, so this should be written as a member function instead.

If you want to extract out a common path it would be better to do it in a way that extracts it from all uses - there are a few other places where we do this kind of texture/renderbuffer fetch. I think the better common code to extract out would be to include the calls to `glGetNamedFramebufferAttachmentParameterivEXT` to get the object name and type. So you would then instead have a function that returns the ID of a given framebuffer's attachment. The access of `m_ResourceUses` wouldn't be inside the function since it may not always be `ResourceUsage::Clear`, but it has the ID to use directly so it's only one line.